shaft_element.py

```diff
@@ -1,478 +1,461 @@
+        pass
+    
+    def K(self):
+        pass
+
+class ShaftElement(Element):
+    r"""A shaft element.
+    This class will create a shaft element that may take into
+    account shear, rotary inertia an gyroscopic effects.
+    The matrices will be defined considering the following local
+    coordinate vector:
+    .. math:: [x_1, y_1, \alpha_1, \beta_1, x_2, y_2, \alpha_2, \beta_2]^T
+    Where :math:`\alpha_1` and :math:`\alpha_2` are the bending on the yz plane and
+    :math:`\beta_1` and :math:`\beta_2` are the bending on the xz plane.
+    Parameters
+    ----------
+    L : float
+        Element length.
+    i_d : float
+        Inner diameter of the element.
+    o_d : float
+        Outer diameter of the element.
+    material : ross.material
+        Shaft material.
+    n : int, optional
+        Element number (coincident with it's first node).
+        If not given, it will be set when the rotor is assembled
+        according to the element's position in the list supplied to
+        the rotor constructor.
+    axial_force : float
+        Axial force.
+    torque : float
+        Torque.
+    shear_effects : bool
+        Determine if shear effects are taken into account.
+        Default is False.
+    rotary_inertia : bool
+        Determine if rotary_inertia effects are taken into account.
+        Default is False.
+    gyroscopic : bool
+        Determine if gyroscopic effects are taken into account.
+        Default is False.
+    Returns
+    -------
+    Attributes
+    ----------
+    Poisson : float
+        Poisson coefficient for the element.
+    A : float
+        Element section area.
+    Ie : float
+        Ie is the second moment of area of the cross section about
+        the neutral plane Ie = pi*r**2/4
+    phi : float
+        Constant that is used according to [1]_ to consider rotary
+        inertia and shear effects. If these are not considered phi=0.
+    References
+    ----------
+    .. [1] 'Dynamics of Rotating Machinery' by MI Friswell, JET Penny, SD Garvey
+       & AW Lees, published by Cambridge University Press, 2010 pp. 158-166.
+    Examples
+    --------
+    >>> from ross.materials import steel
+    >>> le = 0.25
+    >>> i_d = 0
+    >>> o_d = 0.05
+    >>> Euler_Bernoulli_Element = ShaftElement(le, i_d, o_d, steel,
+    ...                                        shear_effects=False, rotary_inertia=False)
+    >>> Euler_Bernoulli_Element.phi
+    0
+    >>> Timoshenko_Element = ShaftElement(le, i_d, o_d, steel,
+    ...                                   rotary_inertia=True,
+    ...                                   shear_effects=True)
+    >>> Timoshenko_Element.phi
+    0.08795566502463055
+    """
+
+    def __init__(
+        self,
+        L,
+        i_d,
+        o_d,
+        material,
+        n=None,
+        axial_force=0,
+        torque=0,
+        shear_effects=True,
+        rotary_inertia=True,
+        gyroscopic=True,
+        shear_method_calc=None,

```

Comment:
It would be better to leave one of the methods as the default:
`shear_method_calc='cowper'`

Reply:
These changes are show in this commit [here](https://github.com/rodrigomoliveira1/ross/commit/3a53437df3e433bd71bbaf316b06eac3dc14e561)